at_secondary/at_secondary_server/lib/src/verb/handler/scan_verb_handler.dart

```diff
@@ -122,27 +133,33 @@ class ScanVerbHandler extends AbstractVerbHandler {
           key.toString().startsWith('privatekey:') ||
           key.toString().startsWith('public:_') ||
           key.toString().startsWith('private:'));
-      keyString = keys.toString();
+      keysList = keys;
     } else {
-      //When pol is performed, display keys that are private to the atsign.
+      // When pol is performed, display keys that are private to the atsign.
       if (atConnectionMetadata.isPolAuthenticated) {
         keys.removeWhere((test) =>
             (test
                     .toString()
                     .startsWith('${atConnectionMetadata.fromAtSign}:') ==
                 false) ||
             test.toString().startsWith('public:_'));
-        keyString = keys
-            .toString()
-            .replaceAll('${atConnectionMetadata.fromAtSign}:', '');
+        for (var key in keys) {
+          var modifiedKey =
+              key.replaceAll('${atConnectionMetadata.fromAtSign}:', '');
```

Comment:
please explain *why* this replacement is being done

Reply:
@VJag per discussion yesterday about keys, this code is a good example of why we shouldn't allow any of our delimiters in keys - ':', '@', etc